ci/src/main/java/ci/ContinuousIntegrationServer.java

```diff
@@ -25,81 +25,138 @@
 import java.io.InputStreamReader;
 import java.io.InputStream;
 import java.lang.ProcessBuilder;
+import java.nio.Buffer;
 import java.util.Enumeration;
 import org.json.*;
+
+import java.io.FileReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Skeleton of a ContinuousIntegrationServer which acts as webhook
  * See the Jetty documentation for API documentation of those classes. ok
  */
 public class ContinuousIntegrationServer extends AbstractHandler {
 
+    static int port = 8080;
+
     public void handle(String target,
                        Request baseRequest,
                        HttpServletRequest request,
                        HttpServletResponse response)
             throws IOException, ServletException {
 
-        String githubEvent = request.getHeader("X-Github-Event");
-        System.out.println(githubEvent);
-        switch(githubEvent){
-            case "push":
-            	
-            	JSONObject payload = new JSONObject(request.getParameter("payload"));
-            	JSONObject headCommit = (JSONObject) payload.get("head_commit");
-            	JSONObject author = (JSONObject) headCommit.get("author");
-            	
-            	
-            	String branchName = (String) payload.get("ref");
-            	branchName = branchName.replaceAll("refs/heads/", "");
-            	String id = (String) headCommit.get("id");
-            	String email = (String) author.get("email");
-            	
-            	
-
-                // here you do all the continuous integration tasks
-                // for example
-                // 1st clone your repository
-                PushTester pushTester = new PushTester();
-
-                File localDirectory = new File("GitPull/");
-
-                Git git = GitConnector.cloneRepo("https://github.com/DD2480-Group26/DD2480-CI.git", localDirectory);
-                GitConnector.gitPull(localDirectory, branchName);
-                GitConnector.checkoutToBranch(localDirectory, "origin/" + branchName);
-
-                PushTester pt = new PushTester();
-                PushStatus pushStatus = pt.getPushStatus(localDirectory);
-
-                // 2nd compile the code
-                pushTester.fileExecuter(localDirectory);
-
-                response.getWriter().println("CI job Done");
-
-                //Delete the directory
-                git.getRepository().close();
-                GitConnector.deleteDirectory(localDirectory);
-                localDirectory.delete();
-                break;
-            case "issues":
-                // DO issues action
-                System.out.println("Issues");
-            default:
-                // DO default actions
-                System.out.println("No event match for " + githubEvent );
-                break;
+        String url = request.getRequestURL().toString();
```

Comment:
maybe you can separate your code into several functions or in a separate class.

Reply:
don't think that a new class makes a lot of sense, but functions for the handler could be good.